src/components/select-field/BaseSelectField.js

```diff
@@ -469,7 +480,7 @@ class BaseSelectField extends React.Component<Props, State> {
 
             const isSelected = selectedValues.includes(value);
 
-            const isClearOption = shouldShowClearOption && index === 0;
+            const isClearOption = shouldShowClearOption && value === 'clear';
```

Comment:
This change is needed because the original implementation was brittle. There can be items that are index 0 but are not the clear button so we need something a little more robust.

Reply:
Is there a risk that a user-provided item could have the value "clear"?

Reply:
Good call. I just tested it and it looks like if the user provides an item with the value clear, this will not work. We still need the `index === 0` check since we always insert the Clear option at the front of the options array.

Reply:
Do we need the "value" check then?

Reply:
We still need the value check because when we are filtering via the search field, the index === 0 element may not be be the clear option so we don't want to apply the clear option styles on that.